### PR TITLE
completed 5-api/users

### PR DESCRIPTION
**Pull request by EmilesGit**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -24,3 +24,25 @@ describe("GET/api/topics", () => {
...
   });
 });
+
+describe("GET/api/users", () => {
+  test("should return an array of user objects", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        console.log(body);
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: get rid of console logs before you make PRs!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -24,3 +24,25 @@ describe("GET/api/topics", () => {
...
+        expect(Array.isArray(body)).toBe(true);
+        expect(body.length > 0).toBe(true);
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: Whenever you do an assertion that involves for loops you should be more specific about the length of the data array you expect back. You can check your test data so you can say 'I expect the array to have exactly this many items'. Please amend to be specific about the length

File: __tests__/app.test.js [hunk 3]
```diff
@@ -24,3 +24,25 @@ describe("GET/api/topics", () => {
       });
   });
 });
+
+describe("GET/api/users", () => {
+  test("should return an array of user objects", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
```
Comment (Poonam-raj) [inline __tests__/app.test.js:70]: the data array should be coming back on the **key** of 'users' on the key of body

So {body :  { users : [....] } }

File: controllers/app.controller.js [hunk 4]
```diff
@@ -9,3 +9,13 @@ exports.getTopics = (req, res) => {
...
     });
 };
+
+exports.getUsers = (req, res) => {
+  selectUser()
+    .then((result) => {
+      res.status(200).send(result);
+    })
+    .catch((err) => {
+      console.log(err);
```
Comment (Poonam-raj) [inline controllers/app.controller.js]: You can get rid of this catch if you aren't using it

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Just need a bit more specificity in your tests and the data to come back on the correct key, address the comments and then throw another PR this way

File: __tests__/app.test.js [hunk 5]
```diff
@@ -24,3 +24,61 @@ describe("GET/api/topics", () => {
...
+      .get("/api/articles/northcoders")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid input");
+      });
+  });
+});
+
+describe("GET/api/users", () => {
+  test("should return an array of user objects", () => {
```
Comment (pkaodev) [inline __tests__/app.test.js]: specify what status code we are receiving in the test description

File: controllers/app.controller.js [hunk 6]
```diff
@@ -9,3 +13,20 @@ exports.getTopics = (req, res) => {
...
     });
 };
+
+exports.getArticle = (req, res, next) => {
+  const { article_id } = req.params;
+  selectArticle(article_id)
+    .then((result) => {
+      res.status(200).send({ result });
+    })
+    .catch(next);
```
Comment (pkaodev) [inline controllers/app.controller.js:23]: unless we're throwing an error that we need to handle ourselves, we don't a catch block

File: controllers/app.controller.js [hunk 7]
```diff
@@ -9,3 +13,20 @@ exports.getTopics = (req, res) => {
...
+};
+
+exports.getUsers = (req, res, next) => {
+  selectUser()
+    .then((users) => {
+      res.status(200).send({ users });
+    })
+    .catch(next);
```
Comment (pkaodev) [inline controllers/app.controller.js:34]: unless we're throwing an error that we need to handle ourselves, we don't a catch block

**Comment by pkaodev** [comment]
Just a couple small changes to make.

after that merge, no need for another PR.

**Pull request merged**